src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrBack = [];
+  let newState = {};
+
+  Object.assign(newState, state);
+
+  for (const act of actions) {
+    switch (act.type) {
+      case 'addProperties':
+        Object.assign(newState, act.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const i of act.keysToRemove) {
+          delete newState[`${i}`];
```

Comment:
Deleting properties from an object does not require Template strings:
```suggestion
          delete newState[i];
```